src/contrib/msc/core/transform/set_expr_layout.cc

```diff
@@ -48,10 +48,14 @@ LayoutDecision InferLayoutDecision(const Expr& expr, const VarLayoutMap& var_lay
 }
 
 LayoutDecision InferLayoutDecisionAt(const Expr& expr, const VarLayoutMap& var_layout_map,
-                                     size_t index = 0) {
+                                     Expr index = PrimValue::Int64(0)) {
   const auto& nlayouts = InferNLayout(expr, var_layout_map);
   if (nlayouts.IsLeaf()) {
-    return index == 0 ? nlayouts.LeafValue() : LayoutDecision("");
+    auto int_index = Downcast<PrimStructInfo>(index->struct_info_)
+                         ->value.as<IntImm>()
+                         .value_or(Integer(0))
```

Comment:
Implemetations like this actually tries to generalize, but leaves more uncertainties in behavior in the code, e.g. what if we have a Tuple[Tuple[Tensor, Tensor], Tensor]?

It is better to  "do less", but ensure what we can handle is robust. Since over generalize can cause unexpected errors that are harder to manage

Reply:
> Implemetations like this actually tries to generalize, but leaves more uncertainties in behavior in the code, e.g. what if we have a Tuple[Tuple[Tensor, Tensor], Tensor]? 

Here, I maintained the existing behavior where the default indicates that we don't have a known decision, and cannot copy a previous layout decision.  For an unknown index, 

> For layout prop pass, the right scope is to handle handling constant tuple indices well, while leave out other cases as opaque.

I'm not opposed to this handling either.  My aim for this PR was to extend the IR, and to handle cases that could be clearly extended.  Where it wasn't clearly extendable, I marked it as a current limitation of the pass, to only support cases where the exact value of the index is known.

Since there's questions on this implementation, it's definitely not a clear generalization, and I'll update to restrict the usage to known indices.

Reply:
The comment is not about requesting changes. Mainly to illustrate that there is a gap and added burden if we go with the generalized TupleGetItem route. So the suggestion is to introduce a separate operator for such cases.